File: script/download_data.R
# author: Yiki Su
# date: 2020-11-20

"This script downloads a zip file from the given URL and unzip the file. Save the unzipped file into a local file called data.

Usage: download_data.R --url=<url> 

Options:
----url=<url>   ULR to download the zip file
" -> doc

library(docopt)

opt <- docopt(doc)

main <- function(url) {
  # To download the zip file from the URL
  download.file(url, destfile = "zip")
  
  # To unzip the zip file to the data folder
  unzip("zip", exdir = "data")
}


main(opt$url)

File: script/download_data.py
# author: Yiki Su
# date: 2020-11-20

"""The sript downloads the zip file from the web to a local filepath and unzip the file. Save it in a local path.
Usage: downoad_data_failed.py --url=<url> --out_path=<out_path>

Options:
--url=<url>              URL from where to download the zip file
--out_path=<out_path>    Path of where to locally save the file
"""
  
from docopt import docopt
import requests
import os
import pandas as pd
import io
import zipfile
from zipfile import ZipFile

opt = docopt(__doc__)

def main(url, out_path):
  try: 
    request = requests.get(url)
    request.status_code == 200
  except Exception as req:
    print("Website at the provided url does not exist.")
    print(req)
    
  zipDocument = zipfile.ZipFile(io.BytesIO(request.content))
  zipDocument.extractall(path=out_path)

if __name__ == "__main__":
  main(opt["--url"], opt["--out_path"])
